### PR TITLE
BDOG-1148

### PR DESCRIPTION
**Pull request by anshulbajpai**
I'll create a subsequent PR to add scaladoc for methods which will refer to the parts of this README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,151 @@
...
+  path = Path.Directory("accounts/2020").file("summary.txt"),
+  content = "this is the content",
+  retentionPeriod = RetentionPeriod.OneMonth, // defaults to 'object-store.default-retention-period' configuration
+  contentType = Some("plain/text"), // defaults to None
+  owner = "my-service" // defaults to 'appName' configuration
+) // returns Future[Unit]
+```
+
+The above code will put the value of `content` parameter in an object at the path `/my-service/accounts/2020/summary.text` with a retention period of 1 month.
+Bear in mind, that this operation will overwrite the content that already exist at that path.
```
Comment (JasperWagen) [inline README.md:92]: Exists?

File: README.md [hunk 2]
```diff
@@ -1,7 +1,151 @@
...
+client.listObjects(
+  path = Path.Directory("accounts/2020"),
+  owner = "my-service" // defaults to 'appName' configuration
+) // returns Future[ObjectListing]
+```
+
+The above code will list the objects at the path `/my-service/accounts/2020/`.
+
+### Error handling
+Any exception like `uk.gov.hmrc.http.GatewayTimeoutException` or response parsing exception will be returned wrapped inside the failed Future.
```
Comment (JasperWagen) [inline README.md:146]: "Exceptions like `uk.gov.hmrc.http.GatewayTimeoutException` or response parsing exception**s** will..."

or maybe 

"Any exception like `uk.gov.hmrc.http.GatewayTimeoutException` or **a** response parsing exception..."

File: README.md [hunk 3]
```diff
@@ -1,7 +1,151 @@
-
 # object-store-client
```
Comment (colin-lamed) [inline README.md:1]: since this is a public multi-module, you could add the badge: ``

File: README.md [hunk 4]
```diff
@@ -1,7 +1,151 @@
...
 
-This is a placeholder README.md for a new repository
+object-store-client is a Scala client to interact with the [object-store](https://github.com/hmrc/object-store) service. 
+The methods available on the client encapsulates the HTTP interaction with object-store service which makes it easier for the consuming service to interact with object-store service.
```
Comment (JasperWagen) [inline README.md:4]: "on the client encapsulate **~s~**"

File: README.md [hunk 5]
```diff
@@ -1,7 +1,151 @@
...
+
+The operations available via object-store-client are 
+- Put a new document
+- Download a document
+- Delete a document
+- List documents under a path
+
+Read more [here](https://github.com/hmrc/object-store/blob/master/README.md) about object-store.
+
+We also have a working example [hello-world-object-store](https://github.com/hmrc/hello-world-object-store) which shows in detail on how to use the object-store-client.
```
Comment (colin-lamed) [inline README.md:14]: maybe `shows in detail how`?

File: README.md (hunk 4 above)
Comment (JasperWagen) [reply README.md:4]: "object-store service ~which makes~ , making  it easier" perhaps?

File: README.md [hunk 6]
```diff
@@ -1,7 +1,151 @@
...
+  retentionPeriod = RetentionPeriod.OneMonth, // defaults to 'object-store.default-retention-period' configuration
+  contentType = Some("plain/text"), // defaults to None
+  owner = "my-service" // defaults to 'appName' configuration
+) // returns Future[Unit]
+```
+
+The above code will put the value of `content` parameter in an object at the path `/my-service/accounts/2020/summary.text` with a retention period of 1 month.
+Bear in mind, that this operation will overwrite the content that already exist at that path.
+
+The method `putObject` is polymorphic in its parameter `content`. At the moment, we support following types -
```
Comment (colin-lamed) [inline README.md:94]: is it worth saying that we recommend using `Source[ByteString, _]` to avoid loading it all into memory. Or at least for large files?

**Review by colin-lamed — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (JasperWagen) [inline README.md:14]: " ~which shows in detail on how to~" 
"which shows, in detail, how to"
or 
"which shows detail on how to"

File: README.md [hunk 7]
```diff
@@ -1,7 +1,151 @@
...
+
+- Configure object-store HTTP configuration under `microservice.services` block in `application.conf`
+```hocon
+object-store {
+  host = localhost
+  port = 8464
+}
+```
+
+Also, override this configuration in app-config-base so this service could interact with object-store service in deployed environments.
```
Comment (JasperWagen) [inline README.md:46]: "so this service could interact with" change to present tense ie
"so this service can interact with" perhaps

File: README.md [hunk 8]
```diff
@@ -1,7 +1,151 @@
...
+```hocon
+object-store {
+  host = localhost
+  port = 8464
+}
+```
+
+Also, override this configuration in app-config-base so this service could interact with object-store service in deployed environments. 
+
+- Configure Authorization token in `application.conf` which grants permissions to this service to interact with object-store via object-store-client.
```
Comment (JasperWagen) [inline README.md:48]: "Configure Authorization token in `application.conf` which grants" ->
"Configure **the** Authorization token in `application.conf` which grants"

**Review by JasperWagen — APPROVED** [review]
Changes look good to me

**Pull request merged**